packages/aleo-types/src/transaction.rs

```diff
@@ -0,0 +1,77 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use bech32::primitives::decode::CheckedHrpstring;
+use bech32::Bech32m;
+use error_stack::{bail, Report, ResultExt};
+use serde::{Deserialize, Serialize};
+
+use crate::Error;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Transaction {
+    transition_id: String,
+}
+
+impl FromStr for Transaction {
+    type Err = Report<Error>;
+
+    fn from_str(message_id: &str) -> Result<Self, Self::Err>
```

Comment:
This is pretty much the same implementation as Transaction::from_str. Can we unify the logic and make something generic? And also call it something not specific to Aleo? This is just bech32 with a prefix and checksum right?

Reply:
done